presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchUtils.java

```diff
@@ -46,23 +46,28 @@ public static Block serializeObject(Type type, BlockBuilder builder, Object obje
         if (MAP.equals(type.getTypeSignature().getBase()) || ARRAY.equals(type.getTypeSignature().getBase())) {
             throw new IllegalArgumentException("Type not supported: " + type.getDisplayName());
         }
-        return serializePrimitive(type, builder, object);
+        serializePrimitive(type, builder, object);
+        return null;
     }
 
     private static Block serializeStruct(Type type, BlockBuilder builder, Object object)
     {
         if (object == null) {
             requireNonNull(builder, "builder is null");
             builder.appendNull();
-            return builder.build();
+            return null;
         }
 
+        List<Type> typeParameters = type.getTypeParameters();
+
+        BlockBuilder currentBuilder;
+
+        boolean builderSynthesized = false;
         if (builder == null) {
+            builderSynthesized = true;
             builder = type.createBlockBuilder(null, 1);
         }
-
-        BlockBuilder currentBuilder = builder.beginBlockEntry();
-        List<Type> typeParameters = type.getTypeParameters();
+        currentBuilder = builder.beginBlockEntry();
 
         for (int i = 0; i < typeParameters.size(); i++) {
```

Comment:
nit: `for (Type typeParameter : typeParameters) {`

Reply:
nit: `for (Type typeParameter : typeParameters) {`

Reply:
seems difficult to get fieldName using type. More convenient to get fieldName using TypeSignatureParameter.